common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -350,15 +415,27 @@ void closeAndDeletePartitionFilesIfNeeded(
    * up older shuffleMergeId partitions. The cleanup will be executed in a separate thread.
    */
   @VisibleForTesting
-  void closeAndDeletePartitionFiles(Map<Integer, AppShufflePartitionInfo> partitions) {
+  void closeAndDeleteOutdatedPartitions(Map<Integer, AppShufflePartitionInfo> partitions) {
     partitions
       .forEach((partitionId, partitionInfo) -> {
         synchronized (partitionInfo) {
           partitionInfo.closeAllFilesAndDeleteIfNeeded(true);
+          removeAppShufflePartitionInfoFromDB(partitionInfo.appAttemptShuffleMergeId);
```

Comment:
Refactored the signature of closeAndDeleteOutdatedPartitions by adding the AppAttemptShuffleMergeId, since every time it is getting called, there is going to only one call to remove the finalized partitionInfo from DB with the key AppAttemptShuffleMergeId.